comms/EclMeta.cpp

```diff
@@ -111,6 +111,47 @@ void CEclMeta::Update(const std::wstring & xml)
     }
 }
 
+bool CEclMeta::LoadImports(const std::_tstring & path, const WPathVector & folders)
+{
+    bool found = false;
+
+    if (m_masterMeta.size())
+    {
+        for (std::map<std::wstring, StlLinked<CEclMetaData> >::iterator itr = m_masterMeta.begin(); itr != m_masterMeta.end(); ++itr)
+        {
+            if (CEclFile *file = dynamic_cast<CEclFile*>(itr->second.get()))
+            {
+                EclImportMap imports = file->GetImports();
+                if (imports.size())
+                {
+                    for (EclImportMap::iterator itrImport = imports.begin(); itrImport != imports.end(); ++itrImport)
+                    {
+                        std::wstring module = itrImport->second.get()->GetRef();
+                        for (WPathVector::const_iterator itr = folders.begin(); itr != folders.end(); ++itr)
+                        {
+                            std::wstring pathstr = itr->first;
+                            if (clib::filesystem::is_directory(pathstr)) {
+                                boost::filesystem::directory_iterator end_itr;
+                                for (boost::filesystem::directory_iterator itrDir(pathstr); itrDir != end_itr; ++itrDir)
+                                {
+                                    if (clib::filesystem::is_directory(*itrDir) &&
+                                        boost::algorithm::iequals(boost::filesystem::basename(*itrDir), module) &&
+                                        !boost::algorithm::iequals(boost::filesystem::basename(wpathToPath(path).parent_path()), module))
+                                    {
+                                        m_masterMeta[module] = new CEclFolder(*itrDir);
+                                        found = true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return found;
+}
```

Comment:
This looks excessive to me (iterating through all the existing meta items) - is there not an easier way to get the "module" ID from the "path"?

Reply:
@GordonSmith yes, there is a better way to do it. I am working on it now.